test/utils/mocks.js

```diff
@@ -44,12 +68,22 @@ exports.mockNetwork = (calls, done) => {
         finish()
       })
     },
-    start () {
+    start (callback) {
+      setImmediate(() => callback())
+    },
+    findAndConnect (cid, maxProviders, callback) {
+      setImmediate(() => callback())
+    },
+    provide (cid, callback) {
+      setImmediate(() => callback())
```

Comment:
We were missing to call a bunch of these callbacks with just `setImmediate(() => callback)`